File: Java/Statystyka_pozycyjna.java
import java.util.Set;
import java.util.Scanner;
import java.util.TreeSet;

class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        while (input.hasNextLine()) 
        {
            String[] data = input.nextLine().split(" ");
            int j = Integer.parseInt(data[0]) - 1;
            Set<Integer> uniqueNumbers = new TreeSet<>((a, b) -> b - a);
            for (int i = 1; i < data.length; i++)
                uniqueNumbers.add(Integer.parseInt(data[i]));
            Integer[] sortedNumbers = uniqueNumbers.toArray(new Integer[0]);
            if (j < sortedNumbers.length)
                System.out.println(sortedNumbers[j]);
            else
                System.out.println("-");
        }
        input.close();
    }
}

File: Python/Statystyka pozycyjna.py
while True:
    try:
        data = input().split(' ')
        j = int(data[0]) - 1 
        unique_numbers = set(int(data[i]) for i in range(1, len(data)))
        sorted_numbers = sorted(unique_numbers, reverse=True)
        if j < len(sorted_numbers):
            print(sorted_numbers[j])
        else:
            print('-')
    except:
        break